### DISCUSSION TITLE
Bash script for generating grpc files on mac or ubuntu localhost without hacking into Dockerfiles

### DISCUSSION THREAD

**Discussion by reactima** [zitadel/zitadel#4714] [Ideas]
Hi Zitadel team, first time here, and want to say thank you for all your job so far, and share my first time development experience with the source code.  

First impression 
- Having a quick bootstrap from Docker container was really great!
- Roadmap and the level of transparency looks very impressive!
- B2B, B2C cases look very appealing! Please elaborate more over time
- PostgreSQL support is really important  

Second thoughts and issues 
Issue 1.  It's impossible to generate grpc libs from *.proto files in local environment without reverse engineering of Dockerfile from /build folder.  My biggest fear - I won't be able to respond on user issues in self hosted production and contribute to the project without fully generated source code on short notice. Running everything in Dockerfile is not ultimate experience for hacking/studying. Pulling bash commands - too much work.
Issue 2. Building console on localhost seems impossible on the latest versions of nodejs 19.x and typescript 
Issue 3. Too many Active Branches in git repo. Looks like main branch is always behind. It's not clear if you are already fixed previously discussed/discovered issue or actively working on it. It looks like you have you own build system and flow. 

🙏 If possible please share a guideline how to run the project without Dockerfiles. 

It might worth to add bash script for localhost hacking/studying on Mac and Ubuntu. Is it possible at all? 

P.s. I came here after being turned off by Ory 's architecture and complexity of their source code in the search of simpler hackable solution.

Here's the example of Ory Kratos over-engineering which was the number one killer for me and my team members:
https://www.ory.sh/docs/kratos/manage-identities/customize-identity-schema

🙏 Please keep the abstraction level simple if possible. Good hackable examples is much better than sophisticated, "customizable" abstractions from my perspective. Hope Zitadel will be the product which you can be easily integrated into small, mid-size SaaS projects without too much mental efforts and it will eventually funnel you growing clients who wants more - Kubernetes, GCP scalability, observability etc.

**Comment by hifabienne** [comment]
Hei @reactima 
Thank you very much for your feedback we really appreciate it.
We will definitely have a look at it.

**Reply by reactima** [reply]
@hifabienne thank you for dropping by here. i created related discussing in stale buf PR #4553. Hope it helps in the future

**Reply by fforootd** [reply]
It certainly does. I will soon have time again to look into this.
